File: include-axios.js
const zipfile = require('yauzl');
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const vsixFile = 'taskmanager-vscode-1.2.0.vsix';
const tempDir = 'temp_vsix_extract';
const extractDir = path.join(tempDir, 'extension');

// Распаковываем VSIX
execSync(`unzip -q ${vsixFile} -d ${tempDir}`);

// Копируем axios
if (fs.existsSync('node_modules/axios')) {
  const destAxios = path.join(extractDir, 'node_modules', 'axios');
  fs.mkdirSync(path.dirname(destAxios), { recursive: true });
  execSync(`cp -r node_modules/axios ${destAxios}`);
  console.log('Copied axios to package');
}

// Переупаковываем
execSync(`cd ${tempDir} && zip -r -q ../${vsixFile} .`);
execSync(`rm -rf ${tempDir}`);
console.log('VSIX repackaged with axios');

File: include_axios.py
#!/usr/bin/env python3
import zipfile
import os
import shutil

vsix_file = 'taskmanager-vscode-1.2.0.vsix'
temp_dir = 'temp_vsix_extract'
extract_dir = os.path.join(temp_dir, 'extension')

# Распаковываем VSIX
with zipfile.ZipFile(vsix_file, 'r') as z:
    z.extractall(temp_dir)

# Копируем axios
if os.path.exists('node_modules/axios'):
    dest_axios = os.path.join(extract_dir, 'node_modules', 'axios')
    os.makedirs(os.path.dirname(dest_axios), exist_ok=True)
    if os.path.exists(dest_axios):
        shutil.rmtree(dest_axios)
    shutil.copytree('node_modules/axios', dest_axios)
    print('Copied axios to package')

# Переупаковываем
with zipfile.ZipFile(vsix_file, 'w', zipfile.ZIP_DEFLATED) as z:
    for root, dirs, files in os.walk(temp_dir):
        for file in files:
            file_path = os.path.join(root, file)
            arcname = os.path.relpath(file_path, temp_dir)
            z.write(file_path, arcname)

# Удаляем временную директорию
shutil.rmtree(temp_dir)
print('VSIX repackaged with axios')

